backend/src/main/kotlin/vegait/rs/osipodgorica/service/CategoryService.kt

```diff
@@ -19,11 +25,42 @@ class CategoryService(val categoryRepo: CategoryRepository, val uploadService: I
         return categoryRepo.save(category)
     }
 
+    fun update(request: UpdateCategoryRequest, id: Long) =
+        runCatching {
+            categoryRepo.findById(id).get()
+        }.mapCatching { oldCategory ->
+            Category(
+                id = id,
+                name = request.name,
+                relativeUrl = request.thumbnail?.storeNewIcon(id, oldCategory.relativeUrl)
+                    ?: oldCategory.relativeUrl,
+            )
+        }.mapCatching { category ->
+            categoryRepo.save(category)
+        }.getOrThrow()
```

Comment:
ne treba nam ovaj exception handling na kraju jer će se prilikom čuvanja baciti exception ako dođe do nekog problema i to će spring dalje ishendlati

Reply:
ne handle-am ja odje exception, to moze da se radi u `.onFailure{}`. Funkcija samo propagira exception kroz `.getOrThrow()`